test/integration/local_lambda/test_end_to_end.py

```diff
@@ -84,148 +70,170 @@ def invoke_function(self, port):
+        cmd = f"--name {image} -d -v {self.path_to_binary}:/local-lambda-runtime-server -p {port}:8080 --entrypoint /local-lambda-runtime-server/{rie} {image_name} {DEFAULT_1P_ENTRYPOINT} main.success_handler"
 
-        r = self.create_container_and_invoke_function(cmd, port)
+        with self.create_container(cmd, image):
+            r = self.invoke_function(port)
         
-        self.assertEqual(b'"My lambda ran succesfully"', r.content)
+            self.assertEqual(b'"My lambda ran succesfully"', r.content)
+
+            # Make sure we can invoke the function twice
+            r = self.invoke_function(port)
+            
+            self.assertEqual(b'"My lambda ran succesfully"', r.content)
 
-        # Make sure we can invoke the function twice
-        r = self.invoke_function(port)
-        
-        self.assertEqual(b'"My lambda ran succesfully"', r.content)
 
-    @parameterized.expand([("x86_64", "8002"), ("arm64", "9002"), ("", "9052")])
-    def test_lambda_function_arn_exists(self, arch, port):
+    def test_lambda_function_arn_exists(self, arch=ARCH, port=PORT):
         image, rie, image_name = self.tagged_name("arnexists", arch)
 
-        cmd = f"docker run --name {image} -d -v {self.path_to_binary}:/local-lambda-runtime-server -p {port}:8080 --entrypoint /local-lambda-runtime-server/{rie} {image_name} {DEFAULT_1P_ENTRYPOINT} main.assert_lambda_arn_in_context"
+        cmd = f"--name {image} -d -v {self.path_to_binary}:/local-lambda-runtime-server -p {port}:8080 --entrypoint /local-lambda-runtime-server/{rie} {image_name} {DEFAULT_1P_ENTRYPOINT} main.assert_lambda_arn_in_context"
 
-        r = self.create_container_and_invoke_function(cmd, port)
+        with self.create_container(cmd, image):
+            r = self.invoke_function(port)
         
-        self.assertEqual(b'"My lambda ran succesfully"', r.content)
+            self.assertEqual(b'"My lambda ran succesfully"', r.content)
 
-    @parameterized.expand([("x86_64", "8003"), ("arm64", "9003"), ("", "9053")])
-    def test_lambda_function_arn_exists_with_defining_custom_name(self, arch, port):
+
+    def test_lambda_function_arn_exists_with_defining_custom_name(self, arch=ARCH, port=PORT):
         image, rie, image_name = self.tagged_name("customname", arch)
 
-        cmd = f"docker run --name {image} --env AWS_LAMBDA_FUNCTION_NAME=MyCoolName -d -v {self.path_to_binary}:/local-lambda-runtime-server -p {port}:8080 --entrypoint /local-lambda-runtime-server/{rie} {image_name} {DEFAULT_1P_ENTRYPOINT} main.assert_lambda_arn_in_context"
+        cmd = f"--name {image} --env AWS_LAMBDA_FUNCTION_NAME=MyCoolName -d -v {self.path_to_binary}:/local-lambda-runtime-server -p {port}:8080 --entrypoint /local-lambda-runtime-server/{rie} {image_name} {DEFAULT_1P_ENTRYPOINT} main.assert_lambda_arn_in_context"
         
-        r = self.create_container_and_invoke_function(cmd, port)
+        with self.create_container(cmd, image):
+            r = self.invoke_function(port)
```

Comment:
Before we had different ports for each test. But I guess if each container is running and being stopped right away now, then we don't need to have separate ports because they won't clash between tests, right?

Reply:
Yes we can use a unified port now